test/capi/capi_utils.hpp

```diff
@@ -152,6 +155,24 @@ static tc_sframe* make_sframe_double(const std::vector<std::pair<std::string, st
     return sf;
   }
 
+static bool check_equality_gl_sframe(
+  const turi::gl_sframe sf_gl, const turi::gl_sframe ref_gl) {
 
+  size_t num_columns_sf = sf_gl.num_columns();
+  size_t num_columns_ref = ref_gl.num_columns();
+
+  if (num_columns_sf == num_columns_ref) {
+    for (size_t column_index=0; column_index < num_columns_sf; column_index++) {
+      // go through all columns and check for sarray equality one by one
+
+      if (!(sf_gl[column_index] == ref_gl[column_index])) {
```

Comment:
I'm not sure the `==` operator on SArray will do what you intend. I think it returns a new `gl_sarray` (containing 0 or 1 for elementwise equality). I'm not sure what that evaluates to in a boolean context in C++ (in Python, it throws an exception if you attempt to evaluate an SArray as a boolean, but I'm guessing it doesn't in C++).

Reply:
Thanks for pointing that out! Fixed now!